.github/workflows/build_wheels.yml

```diff
@@ -103,9 +103,55 @@ jobs:
           name: cibw-wheels-${{ matrix.os }}-${{ strategy.job-index }}
           path: ./wheelhouse/*.whl
 
+
+  check-minimal-build:
+    # this job isn't meant to upload anything, it's only a sanity check
+    name: Check minimal build requirements (${{ matrix.label }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            arch: x86_64
+            cibw_before_build: pip install --only-binary ':all:' 'numpy==1.25.0' 'Cython==3.0.0' 'setuptools==77.0.1' 'pkgconfig==1.5.5'
+            label: 'numpy-min'
+          - os: ubuntu-latest
+            arch: x86_64
+            cibw_before_build: pip install --only-binary ':all:' 'numpy==2.0.0' 'Cython==3.0.0' 'setuptools==77.0.1' 'pkgconfig==1.5.5'
+            label: 'cython-min + numpy 2.0'
+    steps:
+      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          fetch-depth: 0  # get the non-merge commit for PRs
+
+      # Get HDF5 outside of cibuildwheel so that we can set env vars via GITHUB_ENV
+      # inside the HDF5-getting scripts and have them be in effect for the
+      # cibuildwheel step.
+
+      # Cache HDF5
+      - name: Cache built HDF5
+        uses: actions/cache@0400d5f644dc74513175e3cd8d07132dd4860809 # v4.2.4
+        with:
+          key: ${{ matrix.os }}-${{ matrix.arch }}-6
+          path: cache/hdf5
```

Comment:
I think this cache step is doing nothing here - in the main `build_wheels` job it caches HDF5 built on Windows & Mac, but this job only runs on Linux, and it gets HDF5 from the container image we point cibuildwheel at.